processor/src/operations/io_ops.rs

```diff
@@ -123,6 +128,9 @@ impl Process {
         // update the stack state
         self.stack.shift_left(1);
 
+        self.decoder
+            .set_user_op_helpers(Operation::MLoad, &word[0..1]);
```

Comment:
The operation here should probably be `MStore`. Also, similar to `MStoreW`, we should set helper registers to the old value at memory address (the value before the update), and I don't think we need to truncate it. Meaning, we need to populate helper registers with all 4 old values.